serving/src/main/java/ai/djl/serving/http/KServeRequestHandler.java

```diff
@@ -0,0 +1,155 @@
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.serving.http;
+
+import ai.djl.ModelException;
+import ai.djl.modality.Input;
+import ai.djl.modality.Output;
+import ai.djl.ndarray.types.DataType;
+import ai.djl.ndarray.types.Shape;
+import ai.djl.repository.zoo.ZooModel;
+import ai.djl.serving.models.Endpoint;
+import ai.djl.serving.models.ModelManager;
+import ai.djl.serving.util.NettyUtils;
+import ai.djl.serving.wlm.ModelInfo;
+import ai.djl.serving.workflow.Workflow;
+import ai.djl.util.Pair;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.QueryStringDecoder;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/** A class handling inbound HTTP requests for the KServe API. */
+public class KServeRequestHandler extends HttpRequestHandler {
+
+    private static final Pattern PATTERN = Pattern.compile("^/v2([/?].*)?");
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean acceptInboundMessage(Object msg) throws Exception {
+        if (super.acceptInboundMessage(msg)) {
+            FullHttpRequest req = (FullHttpRequest) msg;
+            return PATTERN.matcher(req.uri()).matches();
+        }
+        return false;
+    }
+
+    @Override
+    protected void handleRequest(
+            ChannelHandlerContext ctx,
+            FullHttpRequest req,
+            QueryStringDecoder decoder,
+            String[] segments)
+            throws ModelException {
+        HttpMethod method = req.method();
+
+        if (HttpMethod.GET.equals(method) && isKServeDescribeModelReq(segments, method)) {
+            handleKServeDescribeModel(ctx, segments);
+        }
+    }
+
+    private boolean isKServeDescribeModelReq(String[] segments, HttpMethod method) {
+        return HttpMethod.GET.equals(method) && segments.length == 4
+                || (segments.length == 6 && "version".equals(segments[4]))
+                        && "v2".equals(segments[1])
+                        && "models".equals(segments[2]);
+    }
+
+    private void handleKServeDescribeModel(ChannelHandlerContext ctx, String[] segments) {
+        String modelName = segments[3];
+        String modelVersion = null;
+        if (segments.length > 4) {
+            modelVersion = segments[5];
+        }
+
+        ModelManager modelManager = ModelManager.getInstance();
+        Map<String, Endpoint> endpoints = modelManager.getEndpoints();
+
+        Endpoint endpoint = endpoints.get(modelName);
+        if (endpoint == null) {
+            sendModelNotFoundError(modelName, modelVersion, ctx);
```

Comment:
Just throw ModelNotFoundException

Reply:
Oh, Their protocol wants to return a json for error response. 

```
metadata_model_error_response =
{
  "error": $string
}
```

https://github.com/kserve/kserve/blob/master/docs/predict-api/v2/required_api.md#model-metadata-response-json-error-object